src/Graphics/SpriteBatch.cs

```diff
@@ -1064,79 +1073,30 @@ byte effects
 				FlushBatch();
 			}
 
-			fixed (VertexPositionColorTexture4* sprite = &vertexInfo[numSprites])
-			{
-				float cornerX = -originX * destinationW;
-				float cornerY = -originY * destinationH;
-				sprite->Position0.X = (
-					(-rotationSin * cornerY) +
-					(rotationCos * cornerX) +
-					destinationX
-				);
-				sprite->Position0.Y = (
-					(rotationCos * cornerY) +
-					(rotationSin * cornerX) +
-					destinationY
-				);
-				cornerX = (1.0f - originX) * destinationW;
-				cornerY = -originY * destinationH;
-				sprite->Position1.X = (
-					(-rotationSin * cornerY) +
-					(rotationCos * cornerX) +
-					destinationX
-				);
-				sprite->Position1.Y = (
-					(rotationCos * cornerY) +
-					(rotationSin * cornerX) +
-					destinationY
-				);
-				cornerX = -originX * destinationW;
-				cornerY = (1.0f - originY) * destinationH;
-				sprite->Position2.X = (
-					(-rotationSin * cornerY) +
-					(rotationCos * cornerX) +
-					destinationX
-				);
-				sprite->Position2.Y = (
-					(rotationCos * cornerY) +
-					(rotationSin * cornerX) +
-					destinationY
-				);
-				cornerX = (1.0f - originX) * destinationW;
-				cornerY = (1.0f - originY) * destinationH;
-				sprite->Position3.X = (
-					(-rotationSin * cornerY) +
-					(rotationCos * cornerX) +
-					destinationX
-				);
-				sprite->Position3.Y = (
-					(rotationCos * cornerY) +
-					(rotationSin * cornerX) +
-					destinationY
-				);
-				fixed (float* flipX = &CornerOffsetX[0]) {
-				fixed (float* flipY = &CornerOffsetY[0]) {
-					sprite->TextureCoordinate0.X = (flipX[0 ^ effects] * sourceW) + sourceX;
-					sprite->TextureCoordinate0.Y = (flipY[0 ^ effects] * sourceH) + sourceY;
-					sprite->TextureCoordinate1.X = (flipX[1 ^ effects] * sourceW) + sourceX;
-					sprite->TextureCoordinate1.Y = (flipY[1 ^ effects] * sourceH) + sourceY;
-					sprite->TextureCoordinate2.X = (flipX[2 ^ effects] * sourceW) + sourceX;
-					sprite->TextureCoordinate2.Y = (flipY[2 ^ effects] * sourceH) + sourceY;
-					sprite->TextureCoordinate3.X = (flipX[3 ^ effects] * sourceW) + sourceX;
-					sprite->TextureCoordinate3.Y = (flipY[3 ^ effects] * sourceH) + sourceY;
-				}}
-				sprite->Position0.Z = depth;
-				sprite->Position1.Z = depth;
-				sprite->Position2.Z = depth;
-				sprite->Position3.Z = depth;
-				sprite->Color0 = color;
-				sprite->Color1 = color;
-				sprite->Color2 = color;
-				sprite->Color3 = color;
-			}
-
 			if (sortMode == SpriteSortMode.Immediate)
 			{
+				// Draw a single sprite right now.
+				fixed (VertexPositionColorTexture4* sprite = &vertexInfo[numSprites])
+				{
```

Comment:
I believe you can avoid pinning the pointer here by using a ref in UpdateVertexInfo()
    UpdateVertexInfo(
                    ref vertexInfo[numSprites],
```

```